### PR TITLE
Define enums for states and handle rebuild state in moac

### PR DESCRIPTION
**Pull request by jkryl**
It also contains fixes for all remaining semistandard issues found in moac source code. For clarity style changes are in a separate commit.

Unit tests for the new feature are quite detailed. Though, thorough testing of the feature in k8s cluster is on todo list and will make sense once the rebuild feature in mayastor is pushed to master branch.

### CODE REVIEW

File: doc/quick.md [hunk 1]
```diff
@@ -172,7 +172,31 @@ This has steps have been tested on:
...
+      Reason:
+      Replicas:
+        Node:  node1
+        Pool:  pool
+        Uri:   bdev:///21d56e09-5b78-11e9-905a-589cfc0d76a7
+      Size:    1073741824
+      State:   online
+    ```
+
+9.  Deploy a pod with fio tool which will be using the PV:
```
[the remark this answers is not in the document]
Comment (jkryl) [reply doc/quick.md:199]: right. I will fix it 👍

File: csi/moac/volume.js [hunk 2]
```diff
@@ -96,14 +96,229 @@ class Volume {
...
+      .length;
+    if (rebuildCount > 0) {
+      this._setState('degraded');
+      return;
+    }
+
+    assert(onlineCount >= this.replicaCount);
+    this._setState('healthy');
+
+    // If we have more online replicas then we need to, then remove one.
```
[the remark this answers is not in the document]
Comment (jkryl) [reply csi/moac/volume.js:223]: it's "repl" parameter in storage class.

File: csi/moac/volume.js [hunk 3]
```diff
@@ -96,14 +96,229 @@ class Volume {
...
+      return {
+        uri: ch.uri,
+        state: ch.state,
+        replica: Object.values(self.replicas).find((r) => r.uri === ch.uri)
+      };
+    });
+    // add newly found replicas to the nexus (one by one)
+    const newReplica = Object.values(this.replicas).filter(
+      (r) => !r.isOffline() && !children.find((ch) => ch.replica === r)
+    )[0];
```
[the remark this answers is not in the document]
Comment (jkryl) [reply csi/moac/volume.js:174]: correct. If there is no match, then filter returns empty array and indexing to empty array returns `undefined`. `undefined` values the `false` in subsequent if condition.

File: mayastor/src/bdev/nexus/nexus_rpc.rs [hunk 4]
```diff
@@ -50,13 +60,34 @@ pub(crate) fn register_rpc_methods() {
                 .map(|nexus| RpcNexus {
                     uuid: name_to_uuid(&nexus.name).to_string(),
                     size: nexus.size(),
-                    state: nexus.state.to_string(),
+                    state: match nexus.state {
```
[the remark this answers is not in the document]
Comment (jkryl) [reply mayastor/src/bdev/nexus/nexus_rpc.rs:63]: so there are two places where this conversion is done. This particular place will go away for sure as it is implementation of jsonrpc methods that was replaced by grpc but hasn't been removed yet.

File: csi/moac/volume.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply csi/moac/volume.js:223]: right, yep makes sense thank you

File: csi/moac/volume.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply csi/moac/volume.js:174]: :+1

**Pull request merged**